### ISSUE TITLE
Possibly writing up the steps to build and compile maps

### ISSUE THREAD

**Issue by thrownewexception** [Nukem9/LinkerMod#3]
I'd love to do this and or add on to make things more user friendly. If you have the time to provide a simple steps list I can improve on it.

**Comment by SE2Dev** [comment]
I haven't finished the setup tool yet - all of the required files aren't yet included. The reason it isn't user friendly is because it's not done.

**Comment by thrownewexception** [comment]
I noticed files were missing, most notably the steam api and a steam_appid.txt to allow the setup to actually run, I managed to compile everything and run setup. Are you interested in a pull that fixes the directory structure for d3dlib that you just added? the paths seem wrong initially...

**Comment by SE2Dev** [comment]
The SteamAPI files can't be redistributed as per the Steamworks ToS, I may eventually drop it as a requirement and just use registry keys. 

As for D3DBSP_Lib, which branch are you compiling? I could've sworn the paths were correct last time I checked.

**Comment by thrownewexception** [comment]
Yep just a little but, I adjusted the output dir for d3dbsplib and them adjusted the paths in stdafx for the required mods and added an include to path_mod. I know about the steam API :P 

Possible route would be to use the registry to get the steam install path and parse the "config\config.vdf" for library folders, then you could easily replace getting BO1's install path for different configurations.

**Comment by SE2Dev** [comment]
The D3DBSP_Lib submodule should be resolve to a valid path.

**Comment by thrownewexception** [comment]
Nice, looking good 😎

**Comment by thrownewexception** [comment]
Ayy mod compiled and loaded http://imgur.com/SDgCWTY

**Comment by thrownewexception** [comment]
So when ever you're ready to push them xanims ;)

**Comment by SE2Dev** [comment]
The xanims should be fine - all dog, standard zombie anims are kept in common_zombie.ff

**Comment by thrownewexception** [comment]
And common_zombie is loaded by default?

**Comment by SE2Dev** [comment]
As long as your map name begins with `zombie_` and the zone has `ignore,code_post_gfx
ignore,common_zombie` the anims should be loaded from common_zombie.ff

**Comment by thrownewexception** [comment]
Took all day to figure that out, had a hunch xD

**Comment by thrownewexception** [comment]
What we had to do was copy common_zombie to the mod dir and rename it to mod.ff. That got us the huds working and zombie anims, whats still missing is the off perk models, as the powered on ones are in the common_zombie :)

**Comment by SE2Dev** [comment]
I highly advise against renaming common_zombie and moving it to the mod dir as it would effectively make it impossible to distribute your mod without distributing actual game files along with it.

**Comment by thrownewexception** [comment]
No I know
haha I already fixed it and don't need to do that anymore.

**Comment by thrownewexception** [comment]
@SE2Dev Any tips on how to make the lighting better? Everything sorta glows and such...

**Comment by SE2Dev** [comment]
Try turning the light intensity down - or disabling HDR in launcher

**Comment by thrownewexception** [comment]
@SE2Dev Thanks and the only other thing would be gibbing crash's the game Lol

**Comment by SE2Dev** [comment]
Make sure you assign the proper physic preset to the spawned gib models.

It should be "gib"

**Comment by thrownewexception** [comment]
Hey thanks again, could you describe the process for setting up the mod in the custom maps tab? I assumed a mod.arena but the waw ones we copied crash the game. I already saw the description.txt. @SE2Dev

**Comment by SE2Dev** [comment]
You need to use coopMaps.arena because mods.arena crashes the game.

**Comment by thrownewexception** [comment]
cod2rad exception report: Error Code: Error -1073741819, Details: 

EXCEPTION DETECTED:
EIP: 0x0043C604
EX0: 0x00000001
EX1: 0x77938010


EXCEPTION DETECTED:
EIP: 0x0043C604
EX0: 0x00000001
EX1: 0x77939018

**Comment by thrownewexception** [comment]
Not sure whats causing it but other maps work.

**Comment by thrownewexception** [comment]
Edit: Something to do with you're custom threading, commented out thread patching in cod2rad and fixed the problem.

**Comment by Nukem9** [comment]
If you can, please upload the file somewhere (or privately). Do you know how many tris your map has? It exceeded the 4,000,000 limit I think.

\- or -

You can do it yourself. Change the arbitrary limit:
`BYTE	g_TrisData[4 + 36 * 12 + 4000000]; -----> BYTE	g_TrisData[4 + 36 * 12 + 8000000];`
https://github.com/Nukem9/LinkerMod/blob/master/components/cod2rad/threading.cpp#L9

**Comment by thrownewexception** [comment]
No its literally you're threading code :)

**Comment by thrownewexception** [comment]
I commented out this:

//
	// Threading patches (4 threads -> MAX_THREADS threads)
	//
	//PatchThreading();

and it works fine.

**Comment by Nukem9** [comment]
Yes.....I meant that you should try to change `BYTE g_TrisData` and check if it works with threading...

**Comment by thrownewexception** [comment]
Ahh
Give me one second, the map is huge.

**Comment by thrownewexception** [comment]
Yes threading now works just fine. thanks @Nukem9 :)

**Comment by thrownewexception** [comment]
@SE2Dev and or @Nukem9 Is there any way that the mods can increase the max primary lights? It's 255 right now, I assume that the native bo1 tools support more.

**Comment by SE2Dev** [comment]
Issues such as that would be better placed in their own issue. Also the 255 a limit of BO1 itself, not just the tools.

**Comment by thrownewexception** [comment]
Ahh I'm surprised that they didn't increase this in bo1, I thought It was just because we're using cod2rad from codwaw and it just hadn't been changed.

**Comment by SE2Dev** [comment]
255 is still a _lot_ of primary lights considering only 1 is allowed per surface.

**Comment by thrownewexception** [comment]
Agreed, though we're nearing the brush limit :P

**Comment by DidUknowiPwn** [comment]
What possible map could you have created that it's reaching these limits?

**Comment by thrownewexception** [comment]
I'm working with exofile, lilrifa, radimax, and others.
